core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapFactory.java

```diff
@@ -81,19 +82,20 @@ public abstract DataMapBuilder createBuilder(Segment segment, String shardName,
   /**
    * Get the datamap for all segments
    */
-  public Map<Segment, List<CoarseGrainDataMap>> getDataMaps(List<Segment> segments)
-      throws IOException {
+  public Map<Segment, List<CoarseGrainDataMap>> getDataMaps(List<Segment> segments,
+      List<PartitionSpec> partitions) throws IOException {
     Map<Segment, List<CoarseGrainDataMap>> dataMaps = new HashMap<>();
     for (Segment segment : segments) {
-      dataMaps.put(segment, (List<CoarseGrainDataMap>) this.getDataMaps(segment));
+      dataMaps.put(segment, (List<CoarseGrainDataMap>) this.getDataMaps(segment, partitions));
     }
     return dataMaps;
   }
 
   /**
    * Get the datamap for segmentId
    */
-  public abstract List<T> getDataMaps(Segment segment) throws IOException;
+  public abstract List<T> getDataMaps(Segment segment, List<PartitionSpec> partitions)
```

Comment:
suggest to add one more interface instead of modifying existing one

Reply:
done